src/components/molecules/Search/Search.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import css from './index.module.scss'
+import useAuth from '../../../hooks/useAuth'
+import TeamCard from '../../../components/molecules/TeamCard/TeamCard'
+import DefaultImg from '../../../assets/graphics/default.svg'
+import { returnStatus, singleTeam } from '../../../constants/dropdownOptions'
+import Dropdown from '../Dropdown/Dropdown'
+import Container from '../../Layout/Container'
+import Paragraph from '../../atoms/Paragraph'
+
+const Search = () => {
+  const { teams } = useAuth()
+  const [substring, setSubstring] = useState('')
+  const [filteredTeams, setFilteredTeams] = useState(teams)
+  const [selected, setSelected] = useState('')
+  return (
+    <Container>
+      <div className={css.wrapper}>
+        <label htmlFor="search">
+          <Paragraph tag="h1" color="green" title="Wyszukaj drużynę" />
+          <input
+            type="text"
+            id="search"
+            placeholder="Wyszukaj zespół"
+            value={substring}
+            onChange={(e) => {
+              setSubstring(e.target.value)
+              if (e.target.value === '') {
+                setFilteredTeams(teams)
+              } else {
+                setFilteredTeams(
+                  teams.filter(
+                    (team) =>
+                      team.username.toLowerCase().includes(e.target.value.toLowerCase()) &&
+                      (team.idea?.status === selected ||
+                        selected === '' ||
+                        (!team.idea?.status && selected === 'PENDING'))
+                  )
+                )
+              }
+            }}
+          />
+        </label>
+        <Dropdown
+          defaultValue="Wszystkie drużyny"
+          className={css.dropdown}
+          setSelected={setSelected}
+          selected={selected}
```

Comment:
o tutaj idealny przypadek dlaczego warto skorzystac z hook forma :-)
jezeli bedziesz potrzebowal tworzyc bardziej skomplikowane czy po prostu wieksze formularze to polecam
jakas biblioteke np. formik (https://formik.org/) natomiast osobiscie bardziej preferuje reactowego hooka (https://react-hook-form.com/)